### PR TITLE
Fix portal homepage logic

### PR DESCRIPTION
**Pull request by vcai122**

### CODE REVIEW

File: frontend/pages/index.tsx [hunk 1]
```diff
@@ -23,21 +23,20 @@ export const getServerSideProps = async (
...
-    return {
-      redirect: {
-        destination: `/api/user/clear-cookies/?next=${DASHBOARD_ROUTE}`,
-        permanent: false,
-      },
-    }
+  if (
+    res.status === 403 &&
+    (await res.json()).detail ===
+      'Authentication credentials were not provided.'
```
Comment (vcai122) [inline frontend/pages/index.tsx:29]: not too happy about this hard coding but both errors return a 403, need some way to distinguish...

another thought was making a backend route soley to check if the user was logged into our backend, however thought this was overkill...

File: frontend/pages/index.tsx [hunk 2]
```diff
@@ -23,21 +23,20 @@ export const getServerSideProps = async (
...
-    }
-  } else {
-    return {
-      redirect: {
-        destination: `/api/user/clear-cookies/?next=${DASHBOARD_ROUTE}`,
-        permanent: false,
-      },
-    }
+  if (
+    res.status === 403 &&
```
Comment (judtinzhang) [inline frontend/pages/index.tsx:27]: idk Javascript but is there a reason you don't need to await for status but do need to for the json?

**Review by judtinzhang — APPROVED** [review]
LGTM. My only concern here is now that we are catching the 403 errors on backend, we no longer have the stack trace on sentry. Should be fine though since we already know the root problem + we can test on bastion

File: frontend/pages/index.tsx (hunk 2 above)
Comment (vcai122) [reply frontend/pages/index.tsx:27]: no clue tbh, i've just accepted it ig lmao.

**Comment by vcai122** [comment]
> LGTM. My only concern here is now that we are catching the 403 errors on backend, we no longer have the stack trace on sentry. Should be fine though since we already know the root problem + we can test on bastion

we should set up some sort of logging

**Pull request merged**